samples/src/main/java/com/example/bigquery/ExtractTableToJson.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+// [START bigquery_extract_table]
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+
+public class ExtractTableToJson {
+
+  public static void runExtractTableToJson() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "bigquery-public-data";
+    String datasetName = "samples";
+    String tableName = "shakespeare";
+    String bucketName = "my-bucket";
+    String destinationUri = "gs://" + bucketName + "/path/to/file";
+    extractTableToJson(projectId, datasetName, tableName, destinationUri);
+  }
+
+  // Exports datasetName:tableName to destinationUri as raw CSV
+  public static void extractTableToJson(
+      String projectId, String datasetName, String tableName, String destinationUri) {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
```

Comment:
All of the methods that can throw `BigQueryException` need to be in a try/catch.

`BigQueryException` is unfortunately an unchecked exception, so we need to be super careful to show users to catch it (since it's not declared in the method).